src/Flex.php

```diff
@@ -182,6 +182,25 @@ public function activate(Composer $composer, IOInterface $io)
         }
     }
 
+    public function getLockFilePath()
+    {
+        $composerFile = Factory::getComposerFile();
+        $lockFilePath = "json" === pathinfo($composerFile, PATHINFO_EXTENSION)
```

Comment:
looks magic? taken from https://github.com/composer/composer/blob/ea9b7ecbb05c4b4265b1e15d72b2695439fdc12a/src/Composer/Factory.php#L389

Reply:
I haven't keep this code as I don't see why one would not end its `composer.json` file with `.json`. Using `composer.foobar` works (I've tested it), but it does not seem something worth supporting.